lib/exhal/transcoder.ex

```diff
@@ -79,19 +79,21 @@ defmodule ExHal.Transcoder do
   initial_params - the initial params with which the newly extracted info should
     merged.
   src_doc - the document to interpret
+  opts - options for use by modules adopting ExHal.ValueConverterWithOptions behaviour
   """
-  @callback decode!(%{}, ExHal.Document.t) :: %{}
-  @callback decode!(ExHal.Document.t) :: %{}
+  @callback decode!(%{}, ExHal.Document.t, []) :: %{}
```

Comment:
`[]` means only empty lists are acceptable, which not what i think you mean. `[key: any()]` would be better.